src/query/storages/fuse/src/table_functions/fuse_vacuum_drop_aggregating_index.rs

```diff
@@ -0,0 +1,156 @@
+    }
+}
+
+#[async_trait::async_trait]
+impl SimpleTableFunc for FuseVacuumDropAggregatingIndex {
+    fn get_engine_name(&self) -> String {
+        "fuse_vacuum_drop_aggregating_index".to_owned()
+    }
+
+    fn table_args(&self) -> Option<TableArgs> {
+        Some((&self.args).into())
+    }
+
+    fn schema(&self) -> TableSchemaRef {
+        TableSchemaRefExt::create(vec![TableField::new("result", TableDataType::String)])
+    }
+
+    async fn apply(
+        &self,
+        ctx: &Arc<dyn TableContext>,
+        _plan: &DataSourcePlan,
+    ) -> Result<Option<DataBlock>> {
+        let catalog = ctx.get_default_catalog()?;
+        let duration = Duration::days(ctx.get_settings().get_data_retention_time_in_days()? as i64);
+        let retention_time = chrono::Utc::now() - duration;
+        let tenant = ctx.get_tenant();
+        let table = match &self.args.database_table {
+            Some((database_name, table_name)) => Some(
+                catalog
+                    .get_table(&tenant, database_name, table_name)
+                    .await?,
+            ),
+            None => None,
+        };
+        let table_id = table.map(|t| t.get_id());
+
+        let reply = catalog
+            .get_marked_deleted_indexes(&tenant, table_id)
+            .await?;
+
+        info!(
+            "duration: {:?}, retention_time: {:?}, table_id: {:?}, marked_deleted_indexes: {:?}",
+            duration, retention_time, table_id, reply
+        );
+
+        for TableIndexes { table_id, indexes } in reply.table_indexes {
+            let Some(table_meta) = catalog.get_table_meta_by_id(table_id).await? else {
+                // Skip vacuuming indexes of dropped tables - this will be handled by the vacuum drop table operation
+                info!("skip vacuuming indexes of dropped table: {}", table_id);
+                continue;
+            };
+            let table_info = TableInfo::new(
+                Default::default(),
+                Default::default(),
+                TableIdent::new(table_id, table_meta.seq),
+                table_meta.data,
+            );
+            let table = catalog.get_table_by_info(&table_info)?;
+            let indexes_to_be_vacuumed = indexes
+                .into_iter()
+                .filter(|(_, index_meta)| index_meta.dropped_on.as_ref().unwrap() < &retention_time)
+                .map(|(index_id, _)| index_id)
+                .collect::<Vec<_>>();
+            info!(
+                "indexes_to_be_vacuumed for table: {:?}, indexes: {:?}",
+                table_id, indexes_to_be_vacuumed
+            );
+            for index_id in &indexes_to_be_vacuumed {
+                table.remove_aggregating_index_files(*index_id).await?;
+            }
+            catalog
+                .remove_marked_deleted_index_ids(&tenant, table_id, &indexes_to_be_vacuumed)
+                .await?;
+        }
+        let col: Vec<String> = vec!["Ok".to_owned()];
```

Comment:
We can return more information, such as table name, cleaned index name, number, etc